### PR TITLE
Expose Prometheus-scrapable ports to all pods in the cluster

### LINKED ISSUE #371: Detect required connectivity to Prometheus
Pods behind Services with the annotation `prometheus.io/scrape: "true"` should allow in-cluster ingress into port 9090.
If, in addition, the Service has the annotation `prometheus.io/port`, use the specified port instead.

### PR DESCRIPTION
**Pull request by zivnevo**
Fixes #371

### CODE REVIEW

File: tests/sockshop/manifests/26-user-svc.yaml [hunk 1]
```diff
@@ -5,14 +5,19 @@ metadata:
...
   labels:
     name: user
   namespace: sock-shop
 spec:
   ports:
     # the port that this service should serve on
   - port: 80
+    name: http
     targetPort: 80
+  - port: 7070
```
Comment (adisos) [inline tests/sockshop/manifests/26-user-svc.yaml:18]: is it ok that `user-dep` has no such `containerPort`?

File: tests/sockshop/expected_netpol_output.json [hunk 2]
```diff
@@ -309,6 +330,21 @@
                         "name": "rabbitmq"
                     }
                 },
+                "ingress": [
+                    {
+                        "ports": [
+                            {
+                                "protocol": "TCP",
```
Comment (adisos) [inline tests/sockshop/expected_netpol_output.json:337]: btw, noticed that `rabbitmq` svc has `prometheus.io/scrape: 'true'`, but the deployment has `prometheus.io/scrape: "false"`

**Review by adisos — COMMENTED** [review]
Looks good, added just small questions

File: tests/sockshop/manifests/26-user-svc.yaml (hunk 1 above)
Comment (zivnevo) [reply tests/sockshop/manifests/26-user-svc.yaml:18]: Not specifying a containerPort doesn't prevent that port from being exposed.
See [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#container-v1-core)

File: tests/sockshop/expected_netpol_output.json (hunk 2 above)
Comment (zivnevo) [reply tests/sockshop/expected_netpol_output.json:337]: yes, I'm not quite sure what was the user intent in this case, neither how Prometheus will behave.
Doesn't seem to harmful to open the specified port anyway.

**Review by adisos — APPROVED** [review]

**Pull request merged**